hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataTableValidator.java

```diff
@@ -1136,6 +1151,11 @@ private void validateFileSlices(
           mismatch = true;
           break;
         }
+        // test for log files equality
+        if (!fileSlice1.getLogFiles().collect(Collectors.toList()).equals(fileSlice2.getLogFiles().collect(Collectors.toList()))) {
```

Comment:
This introduced a bug that overshadows the logic in `areFileSliceCommittedLogFilesMatching`, causing false positives.

Reply:
This bug is fixed in #12141.